includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -409,31 +409,44 @@ public function __construct(
 		if ( $this->is_saved_cards_enabled() ) {
 			array_push( $this->supports, 'tokenization', 'add_payment_method' );
 		}
+	}
+
+	/**
+	 * Initializes this class's WP hooks.
+	 *
+	 * @return void
+	 */
+	public function init_hooks() {
+		add_action( 'woocommerce_update_order', [ $this, 'schedule_order_tracking' ], 10, 2 );
 
-		add_action( 'woocommerce_update_options_payment_gateways_' . $this->id, [ $this, 'process_admin_options' ] );
-		add_action( 'admin_notices', [ $this, 'display_errors' ], 9999 );
-		add_action( 'woocommerce_order_actions', [ $this, 'add_order_actions' ] );
+		// Only add certain actions/filter if this is the main gateway (i.e. not split UPE).
+		if ( self::GATEWAY_ID === $this->id ) {
+			add_action( 'woocommerce_order_actions', [ $this, 'add_order_actions' ] );
+			add_action( 'woocommerce_order_action_capture_charge', [ $this, 'capture_charge' ] );
+			add_action( 'woocommerce_order_action_cancel_authorization', [ $this, 'cancel_authorization' ] );
 
-		add_action( 'wp_ajax_update_order_status', [ $this, 'update_order_status' ] );
-		add_action( 'wp_ajax_nopriv_update_order_status', [ $this, 'update_order_status' ] );
+			add_action( 'wp_ajax_update_order_status', [ $this, 'update_order_status' ] );
+			add_action( 'wp_ajax_nopriv_update_order_status', [ $this, 'update_order_status' ] );
 
-		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts' ] );
-		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts_for_zero_order_total' ], 11 );
-		add_action( 'wp_ajax_create_setup_intent', [ $this, 'create_setup_intent_ajax' ] );
-		add_action( 'wp_ajax_nopriv_create_setup_intent', [ $this, 'create_setup_intent_ajax' ] );
+			add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts' ] );
+			add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts_for_zero_order_total' ], 11 );
```

Comment:
I wonder if these should be moved to the `WC_Payments_Checkout` class. 

When WooPay is enabled with the split-UPE, we need to register a card gateway using the legacy card element and other gateways for additional LPMs using UPE elements: this means that we need to register/enqueue scripts for the `UPE_Split_Payment_Gateway` classes _and_ the `WC_Payment_Gateway_WCPay` class. You can test this by enabling Split UPE checkout and WooPay eligibility from the Dev Tools: now enable WooPay from WCPay settings and proceed into a checkout for an item. If you select the CC gateway and choose "Use a new payment method", you should find that the CC field fails to load, because the scripts have not been enqueued onto the page.

I initially thought that we could resolve this by adding `self::$legacy_card_gateway->init_hooks();` below [this line](https://github.com/Automattic/woocommerce-payments/blob/add/init-hooks/includes/class-wc-payments.php#L496) in `WC_Payments::init()`, but this would have an unintended side-effect of calling all of these hooks twice for the [card gateway instance of `UPE_Split_Payment_Gateway` used for settings](https://github.com/Automattic/woocommerce-payments/blob/add/init-hooks/includes/class-wc-payments.php#L493) and the [legacy card gateway instance of `WC_Payment_Gateway_WCPay` used for the CC checkout gateway](https://github.com/Automattic/woocommerce-payments/blob/add/init-hooks/includes/class-wc-payments.php#L460). 

In another conversation you'd elaborated that the checkout classes (`WC_Payments_Checkout` and `WC_Payments_UPE_Checkout`) should control all checkout page-related stuff and I think these hooks would come underneath that umbrella...but let me know if you disagree or have any better ideas on how to free ourselves from this fiendish foible.